src/core/thread/thread_netif.cpp

```diff
@@ -128,6 +130,7 @@ ThreadError ThreadNetif::Up(void)
 #if OPENTHREAD_ENABLE_DNS_CLIENT
         mDnsClient.Start();
 #endif
+        mChildSupervisor.Start();
```

Comment:
would it be better to only start mChildSupervisor when there is sleepy child? and stop when there is no sleepy child? thus the 1s periodical timer would only be running when required.